src/main/java/taxi/service/DriverServiceImpl.java

```diff
@@ -38,4 +40,12 @@ public Driver update(Driver driver) {
     public boolean delete(Long id) {
         return driverDao.delete(id);
     }
+
+    @Override
+    public Optional<Driver> findByLogin(String login) throws AuthenticationException {
+        return Optional.ofNullable(driverDao.findByLogin(login)
```

Comment:
it's strange to wrap throwing nosuchelement inside another optional
it's service, return Driver not optional